tests/framework/clients/client.go

```diff
@@ -70,13 +71,21 @@ func (c *ClientOperation) Get(namespace string, clientName string) (key string,
 }
 
 // Update updates provided user capabilities
-func (c *ClientOperation) Update(namespace string, clientName string) (caps map[string]string, error error) {
+func (c *ClientOperation) Update(namespace string, clientName string, caps map[string]string) (updatedcaps map[string]string, error error) {
 	context := c.k8sh.MakeContext()
 	logger.Infof("updating the client via CRD")
-	if err := c.k8sh.ResourceOperation("apply", c.manifests.UpdateClient(namespace, clientName)); err != nil {
+	if err := c.k8sh.ResourceOperation("apply", c.manifests.GetClient(namespace, clientName, caps)); err != nil {
 		return nil, err
 	}
 
-	caps, _ = client.AuthGetCaps(context, namespace, "client."+clientName)
-	return caps, nil
+	updatedcaps, _ = client.AuthGetCaps(context, namespace, "client."+clientName)
+	for i := 0; i < 30; i++ {
+		if caps["mon"] == updatedcaps["mon"] {
+			logger.Infof("Finished updating the client via CRD")
+			return updatedcaps, nil
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
Could you add a `logger.Info()` message to indicate we are waiting for the update? It's helps to read the logs.